packages/core/contracts/financial-templates/perpetual-multiparty/ConfigStore.sol

```diff
@@ -0,0 +1,148 @@
+     *        STORE DATA STRUCTURES         *
+     ****************************************/
+
+    // All of the configuration settings available for querying by a perpetual.
+    struct ConfigSettings {
+        // Liveness period (in seconds) for an update to currentConfig to become official.
+        uint256 updateLiveness;
+        // Reward rate paid to successful proposers. Percentage of 1 E.g., .1 is 10%.
+        FixedPoint.Unsigned rewardRatePerSecond;
+        // Bond % (of given contract's PfC) that must be staked by proposers. Percentage of 1, e.g. 0.0005 is 0.05%.
+        FixedPoint.Unsigned proposerBondPct;
+    }
+
+    // Make currentConfig private to force user to call getCurrentConfig, which returns the pendingConfig
+    // if its liveness has expired.
+    ConfigSettings private currentConfig;
+
+    // Beginning on `pendingPassedTimestamp`, the `pendingConfig` can be published as the current config.
+    ConfigSettings public pendingConfig;
+    uint256 public pendingPassedTimestamp;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event ProposedNewConfigSettings(
+        address indexed proposer,
+        uint256 rewardRate,
+        uint256 proposerBond,
+        uint256 updateLiveness,
+        uint256 proposalPassedTimestamp
+    );
+    event ChangedNewConfigSettings(uint256 rewardRate, uint256 proposerBond, uint256 updateLiveness);
+
+    /****************************************
+     *                MODIFIERS             *
+     ****************************************/
+
+    modifier updateConfig() {
+        // Update config settings if possible.
+        _updateConfig();
+        _;
+    }
+
+    constructor(ConfigSettings memory _initialConfig, address _timerAddress) public Testable(_timerAddress) {
+        _validateConfig(_initialConfig);
+        currentConfig = _initialConfig;
+    }
+
+    /**
+     * @notice Returns current config or pending config if pending liveness has expired.
+     * @return ConfigSettings config settings that calling financial contract should view as "live".
+     */
+    function getCurrentConfig() external view override nonReentrantView() returns (ConfigSettings memory) {
+        if (_pendingProposalPassed()) {
+            return pendingConfig;
+        } else {
+            return currentConfig;
+        }
+    }
+
+    /**
+     * @notice Propose new configuration settings. New settings go into effect
+     * after a liveness period passes.
+     * @dev Callable only by owner. Calling this while there is already a pending proposal
+     * will overwrite the pending proposal.
+     */
+    function proposeNewConfig(ConfigSettings memory newConfig) external onlyOwner() nonReentrant() updateConfig() {
+        // Alternatively, we can revert if there is already a pending proposal.
+        // require(pendingPassedTimestamp == 0, "Pending proposal exists");
```

Comment:
should this be commented out?

Reply:
If we enable this, it would mean that you cannot overwrite a proposal once it's in progress. IMO, I think it's better to allow overwrites as long as they restart the timer. WDYT?

Reply:
Sure, I will remove this code but I wanted to put the comment here to draw your attention to it while reviewing. It was an alternate implementation but @mrice32 I do agree with your intuition.

Reply:
Sounds good! Thanks for bringing it up -- hadn't considered overwrites.